### PR TITLE
[#166734366] change toExpressHandler interface

### PR DESCRIPTION
**Pull request by BurnedMarshal**

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please include a description of your PR changes.
  </td>
    </tr>
  </tbody>
</table>



## Affected stories

  * 🐞 [#166734366](https://www.pivotaltracker.com/story/show/166734366): Gli handler express vengono ricreati ad ogni richiesta

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/utils/express.ts [hunk 1]
```diff
@@ -9,10 +9,11 @@ import {
...
  */
-export function toExpressHandler<T>(
-  handler: (req: express.Request) => Promise<IResponse<T>>
-): <P>(req: express.Request, res: express.Response, object?: P) => void {
-  return async (req, res, object) => {
+export function toExpressHandler<T, P>(
+  handler: (req: express.Request) => Promise<IResponse<T>>,
+  object?: P
+): (req: express.Request, res: express.Response) => void {
+  return async (req, res) => {
```
Comment (cloudify) [inline src/utils/express.ts]: don't use `async` here, Express doesn't support async handlers - use then/catch to handle the promise returned by `handler`

Comment (BurnedMarshal) [reply src/utils/express.ts]: fixed in c1c6437

File: src/utils/express.ts [hunk 2]
```diff
@@ -9,20 +9,23 @@ import {
...
-      response.apply(res);
-    }
+export function toExpressHandler<T, P>(
+  handler: (req: express.Request) => Promise<IResponse<T>>,
+  object?: P
+): (req: express.Request, res: express.Response) => void {
+  return (req, res) => {
+    handler
+      .call(object, req)
+      .then(response => {
```
Comment (cloudify) [inline src/utils/express.ts]:
```suggestion
      .catch(ResponseErrorInternal) 
      .then(response => {
        // tslint:disable-next-line:no-object-mutation
        res.locals.detail = response.detail;
        response.apply(res);
     })
```

Comment (BurnedMarshal) [reply src/utils/express.ts]: fixed in c191796

File: src/utils/express.ts [hunk 3]
```diff
@@ -9,20 +9,19 @@ import {
...
-      const response = ResponseErrorInternal(e);
-      // tslint:disable-next-line:no-object-mutation
-      res.locals.detail = response.detail;
-      response.apply(res);
-    }
+export function toExpressHandler<T, P>(
+  handler: (req: express.Request) => Promise<IResponse<T>>,
+  object?: P
+): (req: express.Request, res: express.Response) => void {
+  return (req, res) => {
```
Comment (cloudify) [inline src/utils/express.ts]: remove this curly brace may let you remove the `tslint:disable` directive on the next line

Comment (BurnedMarshal) [reply src/utils/express.ts]: Fixed in e4acea1

**Pull request merged**